oca_port/app.py

```diff
@@ -109,7 +110,20 @@ def __post_init__(self):
         self._check_branch_exists(self.source.ref, raise_exc=True)
         self._check_branch_exists(self.target.ref, raise_exc=True)
         # GitHub API helper
-        self.github = GitHub(self.github_token or os.environ.get("GITHUB_TOKEN"))
+        token = False
+        if os.environ.get("GITHUB_TOKEN"):
+            token = os.environ["GITHUB_TOKEN"]
+        else:
+            token = self.github_token
+            if not token:
+                try:
+                    # get from gh
+                    token = subprocess.check_output(
+                        ["gh", "auth", "token"], text=True
+                    ).strip()
+                except subprocess.SubprocessError:
+                    pass
```

Comment:
This code doesn't match the explanation in README: it takes in priority the token from env, then from the parameter, and from `gh`. I propose the following to give the priority to the parameter first, then env, and `gh`:
```suggestion
        token = self.github_token
        if not token:
            token = os.environ.get("GITHUB_TOKEN")
        if not token:
            try:
                # get from gh
                token = subprocess.check_output(
                    ["gh", "auth", "token"], text=True
                ).strip()
            except subprocess.SubprocessError:
                pass
```

Reply:
We could even consolidate this token resolution in a dedicated method